csrc/runtime/executor_kernel_arg.h

```diff
@@ -22,20 +22,48 @@
 
 namespace nvfuser {
 
+namespace detail {
+template <typename T>
```

Comment:
Double check this is still needed and we otherwise would recurse into the constructor of KernelArgumentHolder.

Reply:
@zasdfgbnm I would value a review of KernelArgumentHolder from you. I could use your advice here, please.